app/client/src/features/dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,169 @@
+    },
+}));
+
+export const DASHBOARD_QUERY = graphql`
+    query DashboardQuery {
+        me {
+            events {
+                edges {
+                    node {
+                        id
+                        title
+                        description
+                        startDateTime
+                        endDateTime
+                        organization {
+                            name
+                        }
+                    }
+                }
+            }
+        }
+    }
+`;
+
+interface Props {
+    queryRef: PreloadedQuery<DashboardQuery>;
+}
+
+export function Dashboard({ queryRef }: Props) {
+    const data = usePreloadedQuery<DashboardQuery>(DASHBOARD_QUERY, queryRef);
+    const listOfEvents = React.useMemo(() => data.me?.events?.edges ?? [], [data.me]);
+    const classes = useStyles();
+    const router = useRouter();
+    const [user, , isLoading] = useUser();
+    const handleNav = (path: string) => () => router.push(path);
+
+    // Verify user is logged in
+    React.useEffect(() => {
+        if (!isLoading && !user) router.push('/');
+    }, [user, router, isLoading]);
+
+    // Store ongoing events
+    const ongoingEvents = React.useMemo(
+        () =>
+            listOfEvents.filter(({ node: event }) => {
+                if (!event.startDateTime || !event.endDateTime) return false;
+                const now = new Date();
+                return isBefore(new Date(event.startDateTime), now) && isAfter(new Date(event.endDateTime), now);
+            }),
+        [listOfEvents]
+    );
+
+    // Store upcoming events
+    const upcomingEvents = React.useMemo(
+        () =>
+            listOfEvents.filter(({ node: event }) => {
+                if (!event.startDateTime) return false;
+                const now = new Date();
+                return isAfter(new Date(event.startDateTime), now);
+            }),
+        [listOfEvents]
+    );
+
+    if (!data) return <Loader />;
+
+    return (
+        <Grid container>
+            <Grid item xs={12} className={classes.item}>
+                <Card className={classes.card}>
+                    <CardContent>
+                        <Typography variant='h6' className={classes.title}>
+                            Current Events
+                        </Typography>
+                        {ongoingEvents.length > 0 ? (
+                            <List>
+                                {ongoingEvents.map(({ node: event }, idx) => {
+                                    let divider = true;
+                                    if (idx === ongoingEvents.length - 1) divider = false;
+                                    return (
+                                        <DashboardEventListItem
+                                            key={event.id}
+                                            event={event}
+                                            ongoing={true}
+                                            divider={divider}
```

Comment:
You can get rid of the above and just inline the logic on divider.
```tsx
divider={idx === ongoingEvents.length - 1}
```

You save on code space this way.

```tsx
 {ongoingEvents.map(({ node: event }, idx) => 
  <DashboardEventListItem
      key={event.id}
      event={event}
      ongoing={true}
      divider={idx === ongoingEvents.length - 1}
/>}
```


Also, this could actually be further broken up I think.
```tsx
// Dashboard.tsx
// you can replace this code with the following
<DashboardEventList events={ongoingEvents} emptyStatusText="No ongoing events to display." />

// DashboardEventList.tsx (a file to be created)
function DashboardEventList({ events, emptyStatusText}: props) {
  if (events.length === 0) return <Typography>{emptyStatusText</Typography>
  return (<List>
  // insert other existing code you already have
  </List>);
}

```

It's not quite copy/paste, but I think you get the idea and it would make this whole file really clean I think.